src/terrain/routes/qms.clj

```diff
@@ -0,0 +1,92 @@
+(ns terrain.routes.qms
+  (:require [terrain.util.config :as config]
+            [terrain.clients.qms :as qms]
+            [terrain.auth.user-attributes :refer [current-user require-authentication]]
+            [terrain.util :refer [optional-routes]]
+            [common-swagger-api.schema :refer [context GET POST DELETE PUT]]
+            [terrain.routes.schemas.qms :as schema]
+            [ring.util.http-response :refer [ok]]))
+
+(defn qms-api-routes
+  []
+  (optional-routes
+   [config/qms-api-routes-enabled]
+
+   (context "/qms" []
+     :tags ["qms"]
+
+     (context "/plans" []
+       (GET "/" []
+         :middleware [require-authentication]
+         :summary schema/GetAllPlansSummary
+         :description schema/GetAllPlansDescription
+         :return schema/PlanListResponse
+         (ok (qms/list-all-plans)))
+
+       (GET "/:plan-id" []
+         :middleware [require-authentication]
+         :summary schema/GetPlanSummary
+         :description schema/GetPlanDescription
+         :path-params [plan-id :- schema/PlanID]
+         :return schema/PlanResponse
+         (ok (qms/single-plan plan-id))))
+
+     ;;; It's /user and not /users since this is for the logged-in user and not
+     ;;; an admin-only lookup, which would require the username to be included
+     ;;; in the path or query-parameters.
+     (context "/user" []
+       (GET "/plan" []
+         :middleware [require-authentication]
+         :summary schema/GetUserPlanSummary
+         :description schema/GetUserPlanDescription
+         :return schema/UserPlanResponse
+         (ok (qms/user-plan current-user)))
```

Comment:
I missed this before, and didn't catch it until I started testing. Sorry about that. `current-user` contains a map with several details about the user:

```suggestion
         (ok (qms/user-plan (:shortUsername current-user))))
```